de.dlr.sc.virsat.model.extension.cef.interfaces.ui/src/de/dlr/sc/virsat/model/extension/cef/interfaces/ui/decorator/InterfaceEndQuantityKindDecorator.java

```diff
@@ -42,10 +42,10 @@ public void decorate(Object element, IDecoration decoration) {
 				if ((iBca != null) && (iBca instanceof DataInterfaceEnd)) {
 					DataInterfaceEnd dIfe = (DataInterfaceEnd) iBca;
 					String iftName = dIfe.getDataInterfaceType() != null ? dIfe.getDataInterfaceType().getName() : "N/A";
-					decorateQuantityAndType(decoration, dIfe.getQuantity(), iftName);
+					decorateQuantityAndType(decoration, dIfe.getQuantityBean().getValue(), iftName);
 				} else if ((iBca != null) && (iBca instanceof AInterfaceEnd)) {
 					AInterfaceEnd aIfe = (AInterfaceEnd) iBca;
-					decorateQuantity(decoration, aIfe.getQuantity());
+					decorateQuantity(decoration, aIfe.getQuantityBean().getValue());

```

Comment:
I think it doesn't save us from an NPE, since `decorateQuantityAndType` wants a primitive long. Let's maybe change argument type to `Long`, and check for null in `decorateQuantityAndType`?